rust/arrow/src/buffer.rs

```diff
@@ -1188,6 +1260,57 @@ impl Drop for SetLenOnDrop<'_> {
     }
 }
 
+/// Creating a `MutableBuffer` instance by setting bits according to the boolean values
+impl std::iter::FromIterator<bool> for MutableBuffer {
+    fn from_iter<I>(iter: I) -> Self
```

Comment:
I prefer this implementation to the 'Buffer' implementation, since it has less pointer juggling.
Originally, it was hair slower...  but with the MutableBuffer improvements, it appears to be a bit faster...  so that's good.
```
group                                         bool-ecd919bde
-----                                         --------------
Buffer::from_iter bool                        1.00      7.2±0.06ms        ? B/sec
MutableBuffer iter bitset                     1.00     62.1±0.05ms        ? B/sec
MutableBuffer::from_iter bool                 1.00      6.3±0.01ms        ? B/sec
```